core/base/assignmentSolver/AssignmentExhaustive.h

```diff
@@ -48,155 +52,9 @@ namespace ttk {
-                                                        unsigned int max_dim) {
-      std::vector<std::vector<int>> allAsgn;
-      std::queue<std::tuple<std::vector<int>, std::vector<int>, bool>> queue;
-
-      std::vector<int> asgn_temp, unasgn_temp;
-      queue.push(std::make_tuple(asgn_temp, unasgn_temp, false));
-
-      while(!queue.empty()) {
-        auto queueElem = queue.front();
-        queue.pop();
-        std::vector<int> asgn = std::get<0>(queueElem);
-        std::vector<int> unasgn = std::get<1>(queueElem);
-        bool toUnasgn = std::get<2>(queueElem);
-        unsigned int ind = asgn.size();
-        for(unsigned int j = 0; j < max_dim + 1; ++j) {
-          std::vector<int> new_asgn(asgn);
-          std::vector<int> new_unasgn(unasgn);
-          if(j < max_dim) {
-            // TODO avoid high complexity of find in vector
-            std::vector<int>::iterator it
-              = std::find(new_asgn.begin(), new_asgn.end(), j);
-            std::vector<int>::iterator it2
-              = std::find(new_unasgn.begin(), new_unasgn.end(), j);
-            if(it == new_asgn.end() and it2 == new_unasgn.end()) { // not found
-              bool addIt = true;
-              if(ind >= min_dim) {
-                for(unsigned int cpt = min_dim; cpt < new_asgn.size(); ++cpt)
-                  if(new_asgn[cpt] > (int)j)
-                    addIt = false;
-                for(unsigned int cpt = 0; cpt < new_unasgn.size(); ++cpt)
-                  if(new_unasgn[cpt] > (int)j)
-                    addIt = false;
-              }
-              if(addIt) {
-                if(not toUnasgn)
-                  new_asgn.push_back(j);
-                else
-                  new_unasgn.push_back(j);
-              } else
-                continue;
-            } else
-              continue;
-
-            unsigned int new_ind = new_asgn.size();
-            if(new_ind == max_dim) {
-              // A new assignment is made here
-              for(auto new_unasgn_elem : new_unasgn)
-                new_asgn.push_back(new_unasgn_elem);
-              // std::sort(new_asgn.begin()+min_dim, new_asgn.end());
-              allAsgn.push_back(new_asgn);
-            } else
-              queue.push(std::make_tuple(new_asgn, new_unasgn, false));
-
-          } else {
-            if(not toUnasgn and ind < min_dim) {
-              new_asgn.push_back(max_dim);
-              queue.push(std::make_tuple(new_asgn, new_unasgn, true));
-            }
-          }
-        }
-      }
-
-      return allAsgn;
-    }
-
-    // std::vector<std::vector<int>>
-    // AssignmentExhaustive::constructAssignments3(int min_dim, int max_dim){
-    std::vector<std::vector<int>> constructAssignments3(unsigned int min_dim,
-                                                        unsigned int max_dim) {
-      std::vector<std::vector<int>> allAsgn;
+    void constructAssignments3(unsigned int min_dim,
```

Comment:
are you sure about the name of that function? (is there a `constructAssignment2`?)
could you please rename it with a more explicit name?